### PR TITLE
BaSP-TG-35: admins add shared components

### PR DESCRIPTION
**Pull request by Maxi-SR**
- Add button component in add, edit and delete.
- Add modal component when you click on button component.
- Add table component.
- Add row component in table component.
- Add React Router to manage routes.

### CODE REVIEW

File: src/Components/Admins/index.js [hunk 1]
```diff
@@ -19,20 +25,81 @@ const Admins = () => {
...
+      setShowModal({
+        ...showModal,
+        [modal]: !showModal[modal]
+      });
+    }
   };
 
   return (
     <section className={styles.container}>
+      <Modal
```
Comment (martumoser) [inline src/Components/Admins/index.js:58]: why you are doing this?

File: src/Components/Admins/index.js [hunk 2]
```diff
@@ -19,20 +25,81 @@ const Admins = () => {
...
+        [modal]: !showModal[modal],
+        [secondModal]: !secondModal[modal]
+      });
+    } else {
+      setShowModal({
+        ...showModal,
+        [modal]: !showModal[modal]
+      });
+    }
   };
```
Comment (martumoser) [inline src/Components/Admins/index.js:54]: I don't understand this logic. toggleModal only has to open and close the modal

File: src/Components/SuperAdmins/index.js [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
+    fetch(`${process.env.REACT_APP_API_URL}/superAdmins`)
       .then((response) => response.json())
       .then((response) => {
         adminsList(response.data);
       });
   }, []);
 
   const handleDelete = async (id) => {
-    await fetch(`${process.env.REACT_APP_API_URL}/admins/${id}`, {
+    await fetch(`${process.env.REACT_APP_API_URL}/superAdmins/${id}`, {
```
Comment (martumoser) [inline src/Components/SuperAdmins/index.js:17]: you don't have to make this changes. Superadmin is the one who see the admin list

File: src/Components/Admins/index.js [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+import Modal from '../Shared/Modal';
 
 const Admins = () => {
+  const history = useHistory();
   const [admins, saveAdmins] = useState([]);
+  const [showModal, setShowModal] = useState({ info: false, delete: false, success: false });
+  const params = useParams();
 
   useEffect(async () => {
     await fetch(`${process.env.REACT_APP_API_URL}/admins`)
```
Comment (martumoser) [inline src/Components/Admins/index.js:16]: Admin should see the list of employees and the projects

**Review by martumoser — CHANGES_REQUESTED** [review]
Why you are adding again the row and table shared component?

File: src/Components/Admins/index.js (hunk 1 above)
Comment (Maxi-SR) [reply src/Components/Admins/index.js:58]: I do not understand the question, why I'm doing what? The return?

File: src/Components/Admins/index.js (hunk 2 above)
Comment (Maxi-SR) [reply src/Components/Admins/index.js:54]: It's depends on how many modals you have to open, sometimes you have to open a modal when other closes. It's explained in the modal PR.

File: src/Components/SuperAdmins/index.js (hunk 3 above)
Comment (Maxi-SR) [reply src/Components/SuperAdmins/index.js:17]: My fault, I was wrong when I changed entities.

File: src/Components/Admins/index.js (hunk 4 above)
Comment (Maxi-SR) [reply src/Components/Admins/index.js:16]: solved

File: src/Components/Admins/form.jsx [hunk 5]
```diff
@@ -36,7 +37,7 @@ const AdminsForm = () => {
   };
 
   const redirect = () => {
-    window.location.assign('/admins');
+    history.push('/admins');
```
Comment (DamianPalavecino) [inline src/Components/Admins/form.jsx:40]:
```suggestion
    history.goBack();
```

**Review by martumoser — CHANGES_REQUESTED** [review]

File: src/Components/Admins/index.js [hunk 6]
```diff
@@ -1,38 +1,149 @@
...
+              ]
+            : ['name', 'lastName', 'email', 'actions']
+        }
+        editItem={editAdmin}
+        handleDelete={openDeleteModal}
+        handleRelatedEntity={handleRelatedEntity}
+      />
+      <div className={styles.selectEntity}>
+        {entity === 'projects' ? (
+          <Button
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Admins/index.js:136]: done

File: src/Components/Admins/index.js [hunk 7]
```diff
@@ -1,38 +1,149 @@
...
+                'description',
+                'clientName',
+                'status',
+                'startDate',
+                'endDate',
+                'employees',
+                'actions'
+              ]
+            : ['name', 'lastName', 'email', 'actions']
+        }
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Admins/index.js:129]: I tried this but I can't, I will try again

[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Admins/index.js:129]: done

**Review by PilarFernandezC — APPROVED** [review]
Looks good to me

**Review by Juanoauil — APPROVED** [review]

**Review by martumoser — APPROVED** [review]

**Comment by EliasBermudez13** [comment]
Looks good, nice Job!!

**Pull request merged**